src/remote.rs

```diff
@@ -1,117 +1,142 @@
-use std::fmt;
+use std::fs;
 use std::path::{Path, PathBuf};
 
 use crate::error::*;
 
 #[derive(Debug)]
 pub struct RemoteAsset {
+    pub filename: String,
     pub origin_path: String,
-    pub label: String,
-    pub response: reqwest::blocking::Response,
-}
-
-impl fmt::Display for RemoteAsset {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{} asset at path, {}, ", self.label, self.origin_path)
-    }
+    pub contents: Vec<u8>,
 }
 
 impl RemoteAsset {
-    pub fn load(origin_path: &str, label: &str) -> Result<RemoteAsset> {
-        match reqwest::blocking::get(origin_path) {
-            Ok(response) => Ok(RemoteAsset {
-                origin_path: origin_path.to_string(),
-                label: label.to_string(),
-                response,
-            }),
+    pub async fn load(origin_path: &str) -> Result<RemoteAsset> {
+        match reqwest::get(origin_path).await {
+            Ok(response) => {
+                let filename = RemoteAsset::filename(origin_path, response.headers())?;
+                println!("filename is: {}", filename);
```

Comment:
definitely should be using a log macro

Reply:
oh good call i will pull this out before merge, this was just me debuggin